File: 0240-Search a 2D Matrix II/0240-Search a 2D Matrix II.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m = matrix.size();
        if (m == 0)
        {
            return false;
        }
        int n = matrix[0].size();
        if (n == 0)
        {
            return false;
        }
        
        // Start from left bottom or right top element
        int row = m - 1, col = 0;
        while (row >= 0 && col < n)
        {
            if (matrix[row][col] == target)
            {
                return true;
            }
            
            if (matrix[row][col] < target)
            {
                ++col;
            }
            else
            {
                --row;
            }
        }
        
        return false;
    }
};

File: 0201-0300/0240-Search a 2D Matrix II/0240-Search a 2D Matrix II.py
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        row = len(matrix) - 1
        col = 0
        while row >= 0 and col < len(matrix[0]):
            if matrix[row][col] < target:
                col += 1
            elif matrix[row][col] > target:
                row -= 1
            else:
                return True
        return False
